src/workerfarm/WorkerFarm.js

```diff
@@ -25,24 +25,46 @@ class WorkerFarm extends EventEmitter {
     this.callQueue = [];
 
     this.localWorker = require(this.options.workerPath);
-    this.remoteWorker = {
-      run: this.mkhandle('run')
-    };
+    this.run = this.mkhandle('run');
 
     this.init(options);
   }
 
+  warmupWorker(method, args) {
+    // Workers have started, but are not warmed up yet.
+    // Send the job to a remote worker in the background,
+    // but use the result from the local worker - it will be faster.
+    if (this.started) {
+      let promise = new Promise((resolve, reject) => {
+        this.addCall(resolve, reject, method, [...args, true]);
+      });
+      promise
+        .then(() => {
+          this.warmWorkers++;
+          if (this.warmWorkers >= this.children.size) {
+            this.emit('warmedup');
+          }
+        })
+        .catch(() => {});
+    }
+  }
+
   mkhandle(method) {
     return function(...args) {
-      return new Promise((resolve, reject) => {
-        this.addCall({
-          method,
-          args: args,
-          retries: 0,
-          resolve,
-          reject
+      // Child process workers are slow to start (~600ms).
+      // While we're waiting, just run on the main thread.
+      // This significantly speeds up startup time.
+      if (this.shouldUseRemoteWorkers()) {
+        return new Promise((resolve, reject) => {
```

Comment:
cool. any reason why the `new Promise` couldn't also go in `addCall`?

Reply:
Ow damn, totally overlooked that one. f24d3f9a25e3803e6a62f0e4787933ce0e6b70a3